Remove sass

JoseeWouters opened the pull request:
## Context
<!--
What do we want to achieve with this PR? Why did we write this code?
-->

* This PR removes all .scss files and replaces them for .css files. The sass build step is also removed and replaced for some new steps.

## Summary

<!--
Attach one of the following labels to the PR: `changelog: bugfix`, `changelog: enhancement`, `changelog: other`, `changelog: non-user-facing`.
If the changelog item is a bugfix, please use the following sentence structure: Fixes a bug where ... would ... (when ...).
If the changelog item is meant for the changelog of another repo, start you changelog item with the repo name between square brackets, for example: * [wordpress-seo-premium] Fixes a bug where ....
If the same changelog item is applicable to multiple changelogs/repos, add a separate changelog item for all of them.
-->
This PR can be summarized in the following changelog entry:

* Replaces sass with css.

## Relevant technical choices:

* A grunt copy task is added to copy the .css files from `src` to `dist` and to add the product version slug. 
* A Postcss task is added for adding the css files from node_modules.

## Test instructions
<!--
Please follow these guidelines when creating test instructions:
- Please provide step-by-step instructions how to reproduce the issue, if applicable.
- Write step-by-step test instructions aimed at non-tech-savvy users, even if the PR is not user-facing.
-->
This PR can be tested by following these steps:

* Checkout this branch and run `yarn`
* Run `grunt build`
* Make sure the correct files and filenames are added to the `dist` folder
* Make sure there are no visual changes in the plugin

**Criteria for merging to `trunk` as per Herre's Slack message:**
* "running both dev and production builds work as expected in regards to all styles, including creating a zip archive."

## UI changes
* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Documentation
* [ ] I have written documentation for this change.

## Quality assurance

* [x] I have tested this code to the best of my abilities
* [ ] I have added unittests to verify the code works as intended

Fixes #

bintzandt commented on Gruntfile.js:
```diff
@@ -108,24 +108,24 @@ module.exports = function( grunt ) {
-			"css/dist/filter-explanation-<%= pluginVersionSlug %>.css": [ "css/src/filter-explanation.scss" ],
-			"css/dist/search-appearance-<%= pluginVersionSlug %>.css": [ "css/src/search-appearance.scss" ],
-			"css/dist/structured-data-blocks-<%= pluginVersionSlug %>.css": [ "css/src/structured-data-blocks.scss" ],
+			"css/src/admin-global.css": [ "css/src/admin-global.scss" ],
+			"css/src/adminbar.css": [ "css/src/adminbar.scss" ],
+			"css/src/alerts.css": [ "css/src/alerts.scss" ],
+			"css/src/dashboard.css": [ "css/src/dashboard.scss" ],
+			"css/src/edit-page.css": [ "css/src/edit-page.scss" ],
+			"css/src/featured-image.css": [ " css/src/featured-image.scss" ],
+			"css/src/inside-editor.css": [ "css/src/inside-editor.scss" ],
+			"css/src/metabox.css": [ "css/src/metabox.scss" ],
+			"css/src/metabox-primary-category.css": [ "css/src/metabox-primary-category.scss" ],
+			"css/src/toggle-switch.css": [ "css/src/toggle-switch.scss" ],
+			"css/src/wpseo-dismissible.css": [ "css/src/wpseo-dismissible.scss" ],
+			"css/src/yst_plugin_tools.css": [ "css/src/yst_plugin_tools.scss" ],
+			"css/src/yoast-extensions.css": [ "css/src/extensions.scss" ],
+			"css/src/yst_seo_score.css": [ "css/src/yst_seo_score.scss" ],
+			"css/src/yoast-components.css": [ "css/src/yoast-components.scss" ],
+			"css/src/filter-explanation.css": [ "css/src/filter-explanation.scss" ],
+			"css/src/search-appearance.css": [ "css/src/search-appearance.scss" ],
+			"css/src/structured-data-blocks.css": [ "css/src/structured-data-blocks.scss" ],
```
Why do we need to keep this mapping in the `Gruntfile`?

JoseeWouters replied:
We don't. Good catch.

bintzandt commented:
Also: Codeclimate seems to have found some stuff. I have looked through it a little and will comment the ones that seem important:
- `@import` prevents parallel downloads, use `<link>` instead.
	> `@import url(../src/icons.css);`
	> Found in css/src/admin-global.css by csslint

	**Edit: now that I think about it, this will probably be fixed by our build tools right?**
- Furthermore: it mostly complains about IDs in CSS selectors, selectors that are defined twice (all occurrences that I could find were inside `@media` queries), the usage of `!important` (thanks WordPress?) and that width and borders can make elements larger than expected 😛

JoseeWouters commented:
The codeclimate stuff is partially fixed by our build tools indeed, all imports are removed there. And the id's in selectors and other stuff, that's all existing code. Don't think we want to solve that in this PR? Could maybe be (partially) fixed with the feature/admin-redesign branch, but not 100% sure about that.

bintzandt commented:
> The codeclimate stuff is partially fixed by our build tools indeed, all imports are removed there. And the id's in selectors and other stuff, that's all existing code. Don't think we want to solve that in this PR? Could maybe be (partially) fixed with the feature/admin-redesign branch, but not 100% sure about that.

Nah, let's not indeed. After I send the comment I realized that it was probably code that you simply copy-pasted from the old situation 😅

JoseeWouters commented:
>     3. Do we still need `sassdash` as a `devDependency`?
> 
>     4. Probably outside the scope of this PR, but good to keep in mind: we still define sass related tasks in https://github.com/Yoast/plugin-grunt-tasks. Once we have removed all sass we should remove the tasks there as well (and make sure to remove the dependencies).

Removed the sass tasks you noticed, but to the other points I don't have the answer.

bintzandt commented:
@JoseeWouters Just a double check: this is a known issue right?
<img width="681" alt="afbeelding" src="https://user-images.githubusercontent.com/8782235/78225360-5f576400-74ca-11ea-980a-91facf1410e6.png">

bintzandt commented:
Acceptance: everything seems to be working fine. Both the dev build and the prod build work. The dev-server also starts as expected.

One remark: some CSS files are not minified after a production build. Examples are the `admin-global`, `adminbar` and `alerts`. I think it would be best if all the CSS files were minified as well.

GuidoOffermans commented:
ACC with @bintzandt 👍

afercia commented:
Re: the codeclimate `csslint` stuff, I'd consider to disable also the fallback colors rule as it relates to very old browsers not supporting `rgba`. See for example codeclimate complaining here:

https://codeclimate.com/repos/54523c9069568068a70b3279/pull/14636
> Fallback background-color (hex or RGB) should precede RGBA background-color.

Will create a new issue.
